scripts/snapshot-tool.py

```diff
@@ -0,0 +1,137 @@
+  --commit COMMIT       Fetch the latest snapshot of the specified commit
+  --update-bundle-pullspecs
+                        Update the pullspecs in the bundle-patch/bundle.env file
+
+                        
+Example:
+
+$ ~/redhat/dist/konflux/konflux/scripts/snapshot-tool.py
+
+Snapshot tempo-t4pmt
+
+COMPONENT                COMMIT DATE           REVISION                                  COMMIT MESSAGE
+tempo-gateway            2025-01-15T13:31:43Z  5326dff5bf2854725b7b085c648082f4acfa5009  Fix CVE-2024-45338 and CVE-2024-45337 (#191)
+tempo-jaeger-query       2025-01-15T13:31:43Z  5326dff5bf2854725b7b085c648082f4acfa5009  Fix CVE-2024-45338 and CVE-2024-45337 (#191)
+tempo-opa                2025-01-15T13:31:43Z  5326dff5bf2854725b7b085c648082f4acfa5009  Fix CVE-2024-45338 and CVE-2024-45337 (#191)
+tempo-operator           2025-01-15T13:31:43Z  5326dff5bf2854725b7b085c648082f4acfa5009  Fix CVE-2024-45338 and CVE-2024-45337 (#191)
+tempo-query              2025-01-15T13:31:43Z  5326dff5bf2854725b7b085c648082f4acfa5009  Fix CVE-2024-45338 and CVE-2024-45337 (#191)
+tempo-tempo              2025-01-15T13:31:43Z  5326dff5bf2854725b7b085c648082f4acfa5009  Fix CVE-2024-45338 and CVE-2024-45337 (#191)
+tempo-bundle             2025-01-15T17:55:36Z  c4ff84d4494165ec0ae54c7ce31fe7e92aef954f  Update bundle refs (#196)
+tempo-bundle-quay        2025-01-15T17:55:36Z  c4ff84d4494165ec0ae54c7ce31fe7e92aef954f  Update bundle refs (#196)
+"""
+from pathlib import Path
+import subprocess
+import json
+import re
+import argparse
+from urllib.request import urlopen, Request
+from datetime import datetime
+
+repository = Path().absolute().name
+application = repository.replace("konflux-", "").replace("opentelemetry", "otel")
+if not repository.startswith("konflux-"):
+    raise Exception("This script must be run from a Konflux data repository.")
+
+def is_latest_creationTimestamp(a, b):
+    """
+    verifies creation timestamp of two resources
+    returns True if resource B is more recent than A
+    """
+    date_a = datetime.fromisoformat(a["metadata"]["creationTimestamp"])
+    date_b = datetime.fromisoformat(b["metadata"]["creationTimestamp"])
+    return date_a < date_b
+
+def get_latest_resource(resources):
+    latest_resource = None
+    for resource in resources["items"]:
+        if not latest_resource or is_latest_creationTimestamp(latest_resource, resource):
+            latest_resource = resource
+    return latest_resource
+
+def get_snapshot(args):
+    if args.snapshot:
+        p = subprocess.run(["kubectl", "get", "snapshot", args.snapshot, "-o", "json"], capture_output=True, text=True, check=True)
+        return json.loads(p.stdout)
+    elif args.commit:
+        p = subprocess.run(["kubectl", "get", "snapshot", "-l", f"appstudio.openshift.io/application={application},pac.test.appstudio.openshift.io/sha={args.commit}", "-o", "json"], capture_output=True, text=True, check=True)
```

Comment:
appstudio.openshift.io/application={application}-main

Reply:
I added a new CLI argument, defaults to `main`